services/blockchain-indexer/shared/dataService/business/schemas.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { requestConnector } = require('../../utils/request');
+
+let allSchemas;
+
+const getSchemasFromNode = async () => {
+	const systemMetadata = await requestConnector('getSystemMetadata');
+	const schemas = await requestConnector('getSchema');
+	return { schemas, systemMetadata };
+};
+
+const getAllSchemas = async () => {
+	if (!allSchemas) {
+		const schemas = await getSchemasFromNode();
+
+		const commandsParamsSchemas = [];
+		const assetsSchemas = [];
+		const eventsSchemas = [];
+
+		schemas.systemMetadata.modules.forEach(module => {
+			module.events.forEach(event => {
+				const formattedEvents = {};
+				formattedEvents.module = module.name;
+				formattedEvents.name = event.name;
+				formattedEvents.schema = event.data;
+				eventsSchemas.push(formattedEvents);
+			});
+
+			module.assets.forEach(asset => {
+				const formattedAssets = {};
+				formattedAssets.module = module.name;
+				formattedAssets.version = asset.version;
+				formattedAssets.schema = asset.data;
+				assetsSchemas.push(formattedAssets);
+			});
+
+			module.commands.forEach(command => {
+				const formattedTxParams = {};
+				formattedTxParams.moduleCommand = String(module.name).concat(':').concat(command.name);
+				formattedTxParams.schema = command.params;
+				commandsParamsSchemas.push(formattedTxParams);
+			});
+		});
+
+		allSchemas = {};
+		Object.entries(schemas.schemas).forEach(([entity, schema]) => allSchemas[entity] = { schema });
```

Comment:
1. Isn't schema already an object? If so, then why are we wrapping it in another object?
2. Could it be converted to use bluebird promise?

Reply:
1. This is because of the response structure.
2. Same as above.